pkg/proto/eth_transaction.go

```diff
@@ -375,8 +400,8 @@ func (tx *EthereumTransaction) EthereumTxType() EthereumTxType {
 // ChainId returns the EIP155 chain ID of the transaction. The return value will always be
 // non-nil. For legacy transactions which are not replay-protected, the return value is
 // zero.
-func (tx *EthereumTransaction) ChainId() *big.Int {
-	return tx.inner.chainID()
+func (tx *EthereumTransaction) ChainId() int64 {
+	return tx.chainID
 }
```

Comment:
You should change return type to `proto.Scheme`. And I think it would be better if you rename this method to `Scheme()`.